### DISCUSSION TITLE
How to catch $event.detail data in another div in Alpine

### DISCUSSION THREAD

**Discussion by mzhasan2004** [livewire/livewire#3811] [1. Help] [answered]
Hello all, I want to catch data from $event.detail in another div and assign it to php variable inside livewire blade page. How do I do it? Can I do it without using template? Why my below code is not working? This is the code that not working, ```<p><strong x-text="$event.detail.id"></strong></p>```. How do I fix below code? Thanks in advance.

```
<div x-data="{id: 0, anotherID: 1}">
                                            <button @click="id++; anotherID++; $dispatch('open-dropdown',{id, anotherID})">Open Dropdown</button>
                                            <p><strong x-text="id"></strong></p>
                                            <p><strong x-text="anotherID"></strong></p>
 </div>

<div x-data="{ open: false }"
                                            x-show="open"
                                            @open-dropdown.window="if ($event.detail.id % 2 != 0) open = true"
                                            @click.away="open = false" class="table-fixed w-full" >
                                            <p><strong x-text="$event.detail.id"></strong></p>
                                                    
</div>
```

**Comment by joshhanley** [comment, marked as answer]
@mzhasan2016 can you explain why you need that to happen and what you are trying to achieve exactly?

Thanks!

**Reply by mzhasan2004** [reply]
@joshhanley I figured it out already. Can you please suggest me an article or tutorial for Livewire CRUD using Alpine.js table which involves template x-for? Or, if you can suggest me an article or tutorial with Alpine.js table that has template x-for with data coming from Livewire that will be great too. The resources are very limited. Thanks a lot.

**Reply by joshhanley** [reply]
@mzhasan2016 the reason you probably won't find resources like that is most people wouldn't do it that way. They would just render the tabling using Livewire and blade. So again, I have to ask why and what are you trying to achieve?

I'm happy to help, but need some background 🙂

**Reply by mzhasan2004** [reply]
@joshhanley thanks a lot for the reply. I have a table in Livewrie. When a button in a row in that table is clicked, then a dropdown with another table will show the items belonging to that row of the table.  The table I have displays folders and each folder has many different files. So when a folder is clicked a dropdown table should appear below which lists all the files in that folder. If you can show me the way how to do it using Alpine that will be great. If you can show other ways that will be great too. Thanks again for your help.

**Reply by mzhasan2004** [reply]
@joshhanley also I have another question. In x-data how to put function and data together? I have a x-data which is working fine, ```                                        <div x-data="{id: 0, anotherID: 1, another: <?php echo $folder->id ?>}">```. But when I add a function called scope to it like this it does not work, ``` <div x-data="scope(),{id: 0, anotherID: 1, another: <?php echo $folder->id ?>}">``` Why? What is the proper way to put function and data together in x-data? An example will be great. Thanks a lot.

**Reply by joshhanley** [reply]
@mzhasan2016 so one of the benefits of Livewire is you can write your tables and views using blade, instead of having to use javascript, so while you could render the table using Alpine, it sort of goes against the reason for Livewire in the first place.

Sometimes though, you do need Alpine to make the interactivity feel quicker, but if you need to get new data from the server anyway, then it's better to just use Livewire.

In your instance of a folders list, how to structure it (and how much is Livewire vs Alpine) will depend on if you are going to send all the contents of all of the folders in once go, or whether you would prefer to progressively load folders contents as they are expanded.

Regarding your second question, I have a quick question first, why are you using raw php echo instead of `{{ $folder->id }}`?

To merge attributes with a function, it depends on what `scope()` is and returns. You could pass your data object into the scope function as a property and merge the two in there. There may also be a way to do it with the spread operator, but I'm not overly familiar with it. I think questions like this are actually better off on the Alpine forums as they can help better with that stuff 🙂

**Reply by mzhasan2004** [reply]
@joshhanley thanks a lot for taking the time to explain about the solutions related to Livewire, Blade and Alpine.js. I used , echo with $folder->id by mistake while coding. For Alpine as you suggested I will post question in their official forum. Thanks a lot again and bye.
